packages/flutter/test/widgets/scrollable_selection_test.dart

```diff
@@ -0,0 +1,641 @@
+      await tester.sendKeyDownEvent(LogicalKeyboardKey.metaLeft);
+      await tester.sendKeyDownEvent(LogicalKeyboardKey.keyC);
+      await tester.sendKeyUpEvent(LogicalKeyboardKey.keyC);
+      await tester.sendKeyUpEvent(LogicalKeyboardKey.metaLeft);
+
+      final Map<String, dynamic> clipboardData = mockClipboard.clipboardData as Map<String, dynamic>;
+      expect(clipboardData['text'], 'em 0It');
+    }, variant: const TargetPlatformVariant(<TargetPlatform>{ TargetPlatform.iOS, TargetPlatform.macOS }));
+
+    testWidgets('can copy off screen selection - non-mac', (WidgetTester tester) async {
+      final ScrollController controller = ScrollController();
+      final FocusNode focusNode = FocusNode();
+      await tester.pumpWidget(MaterialApp(
+        home: SelectionArea(
+          focusNode: focusNode,
+          selectionControls: materialTextSelectionControls,
+          child: ListView.builder(
+            controller: controller,
+            itemCount: 100,
+            itemBuilder: (BuildContext context, int index) {
+              return Text('Item $index');
+            },
+          ),
+        ),
+      ));
+      focusNode.requestFocus();
+      await tester.pumpAndSettle();
+      final RenderParagraph paragraph0 = tester.renderObject<RenderParagraph>(find.descendant(of: find.text('Item 0'), matching: find.byType(RichText)));
+      final TestGesture gesture = await tester.startGesture(textOffsetToPosition(paragraph0, 2) + const Offset(0, 5), kind: ui.PointerDeviceKind.mouse);
+      addTearDown(gesture.removePointer);
+      final RenderParagraph paragraph1 = tester.renderObject<RenderParagraph>(find.descendant(of: find.text('Item 1'), matching: find.byType(RichText)));
+      await gesture.moveTo(textOffsetToPosition(paragraph1, 2) + const Offset(0, 5));
+      await tester.pumpAndSettle();
+      await gesture.up();
+      expect(paragraph0.selections[0], const TextSelection(baseOffset: 2, extentOffset: 6));
+      expect(paragraph1.selections[0], const TextSelection(baseOffset: 0, extentOffset: 2));
+
+      // Scroll the selected text out off the screen.
+      controller.jumpTo(1000);
+      await tester.pumpAndSettle();
+      expect(find.descendant(of: find.text('Item 0'), matching: find.byType(RichText)), findsNothing);
+      expect(find.descendant(of: find.text('Item 1'), matching: find.byType(RichText)), findsNothing);
+
+      // Start copying
+      await tester.sendKeyDownEvent(LogicalKeyboardKey.controlLeft);
+      await tester.sendKeyDownEvent(LogicalKeyboardKey.keyC);
+      await tester.sendKeyUpEvent(LogicalKeyboardKey.keyC);
+      await tester.sendKeyUpEvent(LogicalKeyboardKey.controlLeft);
+
+      final Map<String, dynamic> clipboardData = mockClipboard.clipboardData as Map<String, dynamic>;
+      expect(clipboardData['text'], 'em 0It');
+    }, variant: const TargetPlatformVariant(<TargetPlatform>{ TargetPlatform.android, TargetPlatform.windows, TargetPlatform.linux, TargetPlatform.fuchsia }));
```

Comment:
Nit: Is the only difference between this and the "mac" test above the keyboard keys?  You could also do a `switch` on the defaultTargetPlatform and do the correct keystrokes there in one test.

Reply:
Same for other tests elsewhere.

Reply:
I prefer to do it this way, using the switch of defaultTargetPlatform may create test that can fail on certain platform only, it may be hard to debug locally if you don't have the device

Reply:
I will keep it as is, let me know if you think otherwise

Reply:
I'm on board with keeping it as-is :+1: